### ISSUE TITLE
$query->dateAdd returning date without quotes

### ISSUE THREAD

**Issue by sanderpotjer** [joomla/joomla-cms#22087]
### Steps to reproduce the issue
Came across this while testing the privacy consent plugin. The easiest way is to `$query->dump()` the query in https://github.com/joomla/joomla-cms/blob/3.9-dev/plugins/system/privacyconsent/privacyconsent.php#L513

### Expected result
```
SELECT `r`.`id`,`r`.`user_id`,`u`.`email`
FROM `cxicl_privacy_consents` AS `r`
LEFT JOIN `cxicl_users` AS `u` ON u.id = r.user_id
WHERE `remind` = 0 AND DATE_ADD('2018-09-08 14:36:27', INTERVAL -323 DAY) > `created`
```
So the date in DATE_ADD quoted. 

### Actual result
```
SELECT `r`.`id`,`r`.`user_id`,`u`.`email`
FROM `cxicl_privacy_consents` AS `r`
LEFT JOIN `cxicl_users` AS `u` ON u.id = r.user_id
WHERE `remind` = 0 AND DATE_ADD(2018-09-08 14:36:27, INTERVAL -323 DAY) > `created`
```
Date in DATE_ADD is not quoted. 

### Additional comments
Looks like this issue is caused by this PR https://github.com/joomla/joomla-cms/pull/21844, that removed the quotes around the date in the string returned.

//cc @alikon

**Comment by alikon** [comment]
please test #22088

**Comment by ghost** [comment]
closed as having Pull Request #22028
